# Generated by Django 5.1.7 on 2025-03-18 14:19

import django.contrib.gis.db.models.fields
import django.db.models.deletion
from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('polygons', '0001_initial'),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='InvalidPolygon',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=255)),
                ('coordinates', django.contrib.gis.db.models.fields.PolygonField(srid=4326)),
                ('reason', models.TextField()),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('intersecting_polygons', models.ManyToManyField(related_name='invalid_intersections', to='polygons.polygon')),
            ],
        ),
        migrations.CreateModel(
            name='PolygonUserAssignment',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('assigned_at', models.DateTimeField(auto_now_add=True)),
                ('assigned_by', models.ForeignKey(null=True, on_delete=django.db.models.deletion.SET_NULL, related_name='assigned_polygons', to=settings.AUTH_USER_MODEL)),
                ('polygon', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='polygons.polygon')),
                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.AddField(
            model_name='polygon',
            name='users',
            field=models.ManyToManyField(related_name='polygons', through='polygons.PolygonUserAssignment', to=settings.AUTH_USER_MODEL),
        ),
    ]
